wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -383,60 +399,107 @@ class PlatformRepo @Inject constructor(
                 profiles[user] = profile
             }
 
-            val usernameSearchResults = ArrayList<UsernameSearchResult>()
-            val searchText = text.lowercase()
+            val usernameSearchResults = getFromProfiles(profiles, text.lowercase(), toContactMap, fromContactMap, includeSentPending)
+            usernameSearchResults.orderBy(orderBy)
 
-            for (profile in profiles) {
-                if (profile.value == null) {
-                    // this happens occasionally when calling this method just after sending contact request
-                    // It occurs when calling NotificationsForUserLiveData.onContactsUpdated() after
-                    // sending contact request (even after adding long delay).
-                    continue
-                }
+            Resource.success(usernameSearchResults)
+        } catch (e: Exception) {
+            Resource.error(formatExceptionMessage("search contact request", e), null)
+        }
+    }
+
+    fun observeContacts(text: String, orderBy: UsernameSortOrderBy, includeSentPending: Boolean = false): Flow<List<UsernameSearchResult>> {
```

Comment:
A new method created that's similar to `searchContacts`, but returns a stream of contacts